aten/src/ATen/native/cuda/Reduce.cuh

```diff
@@ -1108,7 +1104,17 @@ ReduceConfig setReduceConfig(const TensorIterator& iter){
     config.output_mult[1] = config.split_output(block_height);
   }
 
-  const int blocks_per_sm = at::cuda::getCurrentDeviceProperties()->maxThreadsPerMultiProcessor / config.num_threads;
+  // Control the number of threadblocks by adjusting the maximum number of
+  // threads per multi-processor.
+  int max_threads_per_mp =
+      at::cuda::getCurrentDeviceProperties()->maxThreadsPerMultiProcessor;
+#ifdef USE_ROCM
+  if (iter.ndim() == 1)
```

Comment:
Can you add more comments on why different thread block size helps with different dimensions ?